### DISCUSSION TITLE
Общие вопросы

### DISCUSSION THREAD

**Discussion by Sulfurite** [ImMALWARE/bash-warp-generator#55] [Q&A] [answered]
1. Долго живет конфиг на гугл шелле?
2. Для каждого юзера отдельная машина, перехват трафика возможен?

**Comment by ImMALWARE** [comment, marked as answer]
1. Навсегда
2. Нет
Туннель идёт не к google cloud shell, а к cloudflare warp. Google cloud shell нужен лишь для того, чтобы обойти блокировку запроса на создание конфига.

**Comment by Tawally** [comment]
Для Андроид-версии AmneziaWG такой конфиг сработает?
Если да, то нужно генерировать новый конфиг, или использовать тот же, что и на ПК? 
Заранее спасибо)

**Comment by ImMALWARE** [comment]
> Для Андроид-версии AmneziaWG такой конфиг сработает?

сработает, тот же конфиг

**Comment by ChernyshellyOfficial** [comment]
@ImMALWARE к чьему айпи мы подключаемся с помощью амнезии? Чья машина - ваша или cloudflare?

**Comment by ImMALWARE** [comment]
> @ImMALWARE к чьему айпи мы подключаемся с помощью амнезии? Чья машина - ваша или cloudflare?

Cloudflare
https://developers.cloudflare.com/warp-client/warp-modes

**Comment by Tawally** [comment]
Спустя какое-то время в Discord не отображаются никакие каналы и сообщения, failed to load messages 
Аудио-связь продолжает работать)

**Comment by Hess118** [comment]
Конфиг подходит для Amnezia VPN? в WG версии нет раздельного туннелирования

P.S вообще конфиг работает и на VPN версии приложения, но раздельное туннелирование не работает на Windows

**Comment by TmKarter** [comment]
А на сколько это секьюрная штука вообще? Я так понимаю, что свой сгенерированный конфиг никому нельзя передавать из-за приват ключа в нем?

И к какому серверу по итогу идет подключение? Можно ли как-то там наш трафик поймать? Или он через эти ключи из конфига шифруется как раз?

Просто способ рабочий вроде, голосовые каналы в дискорде даже работают. Но теперь кажется, что им всякие мошенники могут воспользоваться...

**Comment by Hess118** [comment]
@TmKarter, подключается к серверу Cloudflare WARP, через протокол AmneziaWG. Потому что в стандартном варианте (через WireGuard) он детектится системой DPI и не работает. Никто трафик не поймает и не перехватит

**Comment by StepashKA5** [comment]
В локальной сети два ПК, на обоих выполнил инструкцию - одновременно работает только один. Точнее - оба подключаются, но на одном Дискорд работает, а на другом нет. Этому есть решение?

**Comment by RTechSn** [comment]
Есть ли ограничение по трафику и если да, можно ли добавлять трафик без создания новой конфигурации?

**Comment by ImMALWARE** [comment]
> Конфиг подходит для Amnezia VPN?

Да, но у меня на Windows не заработало, на Android работает

**Comment by ImMALWARE** [comment]
> Спустя какое-то время в Discord не отображаются никакие каналы и сообщения, failed to load messages
> Аудио-связь продолжает работать)

Это проблема точно не с WARP

**Comment by ImMALWARE** [comment]
> А на сколько это секьюрная штука вообще?

https://developers.cloudflare.com/warp-client/warp-modes

> Я так понимаю, что свой сгенерированный конфиг никому нельзя передавать из-за приват ключа в нем?

Вроде есть лимит на пользователей одновременно, но это не точно. Пользователи делились конфигами и вроде всё работало.

> И к какому серверу по итогу идет подключение?

К Cloudflare

> Можно ли как-то там наш трафик поймать?

Весь трафик в интернете зашифрован https

**Comment by ImMALWARE** [comment]
> Есть ли ограничение по трафику

Не было случаев, когда кто-то поймал ограничение. Да и его не должно быть.

**Comment by ImMALWARE** [comment]
> В локальной сети два ПК, на обоих выполнил инструкцию - одновременно работает только один. Точнее - оба подключаются, но на одном Дискорд работает, а на другом нет. Этому есть решение?

Может, неправильно установил AmneziaWG на втором? Или там запущено что-то ещё для обхода блокировок?

**Comment by StepashKA5** [comment]
> Может, неправильно установил AmneziaWG на втором? Или там запущено что-то ещё для обхода блокировок?
>
Решилось путем установки одинакового конфига на оба устройства

**Comment by OjisanKG** [comment]
Что-то инвайты в дискорде не работают через эту схему, пишет что инвайт просрочен или у вас нет разрешения. Попробовал пяток разных инвайтов которые должны работать. Есть какая-то инфа по этому вопросу?

**Comment by ImMALWARE** [comment]
> Что-то инвайты в дискорде не работают через эту схему, пишет что инвайт просрочен или у вас нет разрешения. Попробовал пяток разных инвайтов которые должны работать. Есть какая-то инфа по этому вопросу?

вряд-ли это связано с warp'ом

**Comment by OjisanKG** [comment]
И правда, через десктопное приложение работают. Может конечно совпадение, что в браузере перестали работать после перехода на варп, но вижу это впервые за много лет пользования дискордом через браузер. 
Значит пока перейду на приложение.

**Comment by Pacifist-AVB** [comment]
Не работает на IOS (2 разных Ipad) с WARP.conf (AmneziaWG с другим конфигм, от другого провайдера работает), на ПК и Андроидах (смартфон и тв-бокс - работает без нареканий.

**Comment by OjisanKG** [comment]
Первого конфига хватило на неделю на ростелекоме в Москве, перестал хендшейки принимать. Новый конфиг заработал.

**Comment by Dim1xs** [comment]
а как сделать так чтоб на ютуб ток работало? давно как то сделал, а щас конфиг новый, но я уже не могу найти где были айпи адреса для ютуба

**Comment by RTechSn** [comment]
> а как сделать так чтоб на ютуб ток работало? давно как то сделал, а щас конфиг новый, но я уже не могу найти где были айпи адреса для ютуба

Хорошее решение - поднимать прокси на устройстве, которое подключается к тоннелю и использовать расширение браузера для выборочного направления сайтов через прокси.

**Comment by Dim1xs** [comment]
> > а как сделать так чтоб на ютуб ток работало? давно как то сделал, а щас конфиг новый, но я уже не могу найти где были айпи адреса для ютуба
> 
> Хорошее решение - поднимать прокси на устройстве, которое подключается к тоннелю и использовать расширение браузера для выборочного направления сайтов через прокси.

какие то заморочки. в любом случае я уже нашел тот способ ( для AmneziaWG ):
изменить `⁡AllowedIPs = 0.0.0.0/0, ::/0`
на `AllowedIPs = 8.8.4.0/24, 8.8.8.0/24, 8.34.208.0/20, 8.35.192.0/20, 23.236.48.0/20, 23.251.128.0/19, 34.0.0.0/10, 35.184.0.0/13, 35.192.0.0/14, 35.196.0.0/15, 35.198.0.0/16, 35.199.0.0/17, 35.199.128.0/18, 35.200.0.0/13, 35.208.0.0/12, 64.18.0.0/20, 64.233.160.0/19, 66.102.0.0/20, 66.249.64.0/19, 70.32.128.0/19, 72.14.192.0/18, 74.114.24.0/21, 74.125.0.0/16, 104.132.0.0/23, 104.133.0.0/23, 104.134.0.0/15, 104.156.64.0/18, 104.237.160.0/19, 108.59.80.0/20, 108.170.192.0/18, 108.177.0.0/17, 130.211.0.0/16, 136.112.0.0/12, 142.250.0.0/15, 146.148.0.0/17, 162.216.148.0/22, 162.222.176.0/21, 172.110.32.0/21, 172.217.0.0/16, 172.253.0.0/16, 173.194.0.0/16, 173.255.112.0/20, 192.158.28.0/22, 192.178.0.0/15, 193.186.4.0/24, 199.36.154.0/23, 199.36.156.0/24, 199.192.112.0/22, 199.223.232.0/21, 207.223.160.0/20, 208.65.152.0/22, 208.68.108.0/22, 208.81.188.0/22, 208.117.224.0/19, 209.85.128.0/17, 216.58.192.0/19, 216.239.32.0/19, 216.239.36.0/24, 216.239.38.0/23, 216.239.40.0/22
`

**Comment by Khanalas** [comment]
Как работает генератор конфига? Хотелось бы получить понятие, чтобы самому можно было подкрутить что нужно когда провайдер начнёт блокировать.

**Comment by ImMALWARE** [comment]
> а как сделать так чтоб на ютуб ток работало?

https://t.me/immalware_chat/39978

**Comment by jolywho** [comment]
а как исправить то что профили в дискорде с варпом не прогружаются ?

**Comment by ImMALWARE** [comment]
> а как исправить то что профили в дискорде с варпом не прогружаются ?

ты случайно не менял allowedips?

**Comment by bubba-champion** [comment]
Проблема с профилями в дискорде возникает, если из `Address` и/или `AllowedIPs` убрать IPv6 адреса. Если все вернуть на свои места, то профиля прогружаются без каких-либо проблем

**Comment by bubba-champion** [comment]
Все же проблема с профилями в дискорде возникает даже с дефолтными `Address` и `AllowedIPs`. Проверено при работе через Keenetic

**Comment by frdkss** [comment]
Как ограничить туннель только для дискорда? Просто жестко пингует в играх, я пытался взять айпишники дискорда, но облом происходит на моменте подключения к текстовому/голосовому каналу.

**Comment by ImMALWARE** [comment]
> Как ограничить туннель только для дискорда? Просто жестко пингует в играх, я пытался взять айпишники дискорда, но облом происходит на моменте подключения к текстовому/голосовому каналу.

https://t.me/immalware_chat/250908

**Comment by McLarenGT3** [comment]
Привет! Насколько это вообще безопасно? Весь трафик идет через сервер, который получается после генерации туннеля, или как? И на сколько хватает конфига?

**Comment by ImMALWARE** [comment]
> Весь трафик идет через сервер, который получается после генерации туннеля, или как

Нет, трафик идёт через сервера Cloudflare. Конфиг вечный.

https://immalware.vercel.app/wiki?page=Безопасность_в_сети:Cloudflare_WARP
